rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -417,24 +438,37 @@ def _deserialize_channel_withdraw(
             address: identity for identity, address in identity_address_map.items()
         }
         try:
-            tx_hash, _, log_index = raw_event['id'].split(':')
-            address = to_checksum_address(raw_event['user'])
-            identity_address = inverse_identity_address_map[address]
+            adex_event = self._deserialize_adex_channel_event(
+                raw_event=raw_event,
+                inverse_identity_address_map=inverse_identity_address_map,
+                case='channel_withdraw',
+            )
+            # raw_event id is already checked in `_deserialize_adex_channel_event()`
+            log_index = raw_event['id'].split(':')[-1]
```

Comment:
Why not move log index into the base AdexEvent and not have a double deserialization here? Is it because only the channel withdraw has it and all other's don't?

Another option if that's the case would be to have `_deserialize_adex_channel_event` return `Tuple[Event, Optional[int]]` so that whoever wants to use it, uses it and whoever doesn't simply discards it.